File: simlf/src/basic/sup/univ/sup_univ_clone.py
from sim import Module
from basic.lib.simm import Sim, apply_ops

class SupUnivClone(Module):
    def run_impl(self):
        sim = Sim({"sys_cache": self.cache_dir.sys_dir, "user_cache": self.cache_dir.user_dir})
        univ_to = sim.write_mod(self.config["par_to"], dtype=bool)
        univ_from = sim.load_mod(self.config["par_from"])
        univ_to[sim.start_di:sim.end_di] = univ_from[sim.start_di:sim.end_di]

File: simlf/src/basic/sup/univ/sup_univ_clone.cpp
#include "yao/base_daily.h"
using namespace yao;

namespace {
using namespace std;
class SupUnivClone : public yang::Module {
 public:
  void RunImpl() final {
    auto univ_to = WriteArray<bool>(config<string>("par_to"));
    auto &univ_from = *ReadArray<bool>(config<string>("par_from"));
    univ_to.CopyFrom(univ_from, start_di(), end_di());
  }
};

REGISTER_MODULE(SupUnivClone);
}  // namespace
